client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCacheManager.java

```diff
@@ -202,6 +209,21 @@ public RemoteCacheManager() {
       return createRemoteCache("", forceReturnValue);
    }
 
+   @Override
+   public <K, V> TransactionalRemoteCache<K, V> getTransactionalCache(String cacheName,
```

Comment:
... above in the sense that I'd omit the TransactionalRemoteCacheImpl altogether.

Reply:
the problem is the transaction manager is always there (it defaults to RemoteTM, to keep it symmetric with embedded) but we don't have any per-cache configuration in the client. and I don't want to add it.

Reply:
Okay, then use `tm.getStatus()` in any call. We don't have a TransactionalCache interface and tx impl in embedded mode either, and there was a strong call for parity.

Reply:
That won't work , unfortunately. As an example, you can have non-tx cache invocations inside a transaction:

```
tm.begin()
tx-cache.put(k,v);
non-tx-cache.put(k,v); <= in this case, tm.getStatus() will return an active transaction
```

The way we do it in embedded is to have a null TM for non-tx caches. But we ended with a lot of `if (transactionManager != null)`. I would prefer to split the cache implementation in embedded than merge remove the TransactionalRemoteCache...

Reply:
@pruivo Since you don't have any per-cache configuration there'd be nothing like a non-transactional remote cache. If user wants to do a non-transactional operation when he's in the middle of an active transaction, he'd have to suspend the running tx and resume is afterwards - it's not so much of a hassle.

Reply:
@pruivo Old problem? You'll enlist all caches as either sync or xa, depending on the global configuration. I don't say that you shouldn't have some `getCache(...)` overloads, though I wonder how useful these are when you can create another RCM.

Reply:
```
When creating the {rcm}, it is possible to set the default {tm} and {tx-mode} to use by the {rc}.
However, it is possible to set a different {tm} and {tx-mode} for each cache.
```
that's why I have the getTxCache methods so the user can change the TM or the TxMode per cache.

Reply:
As a user, I don't care too much about sync/xa, I want to call `getCache("foo")` without any overloads and get an instance that I can later use for transactions.
I don't want to care about extra interface and rewrite all signatures in my code from `RemoteCache` to `TransactionalRemoteCache`, just to be able to get the TM.
Least of all, the `Transactional` in `getTransactionalCache` is superfluous - the cache is a concept, it depends on the way how I use it.

Reply:
I disagree. 
IMO, the user should be able to choose to which `TransactionManager` he wants to interact with and how the cache should interact with. It is the same idea as `forceReturnValue` and `getCache(cacheName, forceReturnValues)`.  It isn't fair to make the user create N `RemoteCacheManager` to use the caches differently. 
It also keeps symmetric with embedded mode.

I guess will need a 3rd opinion otherwise we won't move forward. @tristantarrant or @danberindei, any ideas?

Reply:
@pruivo I was never suggesting to remove the option, overloads are completely fine. I just said that as long as global configuration is set correctly, the user should not be forced to use anything but `getCache(name)`.

Reply:
In my view the `getTransactionalCache` methods are introducing the asymmetry.

Reply:
@pruivo I think `getCache(name, forceReturnValues)` makes sense because there's no configuration behind that, either on the client or in the server. If you need some extra configuration for a cache to work as a tx cache, then there's no point in having an overload with an extra parameter that fails when the user calls it with the wrong parameter.

TBH I haven't read most of the PR yet so I don't know where you stand :)

Reply:
> there's no configuration behind that

@danberindei what do you mean by that? https://github.com/infinispan/infinispan/blob/master/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java#L450

Reply:
I forgot that it's in the configuration as well, but my point was that you can enable `forceReturnValues` just by calling `getCache(name, true)`, without having anything about it in the configuration.